dotCMS/src/main/java/com/dotcms/integritycheckers/IntegrityUtil.java

```diff
@@ -243,79 +257,162 @@ public static void unzipFile(InputStream zipFile, String outputDir) throws Excep
+    }
+
+    private static void saveIntegrityDataStatus(IntegrityDataControl integrityDataControl) {
+        final String endpointId = integrityDataControl.getEndpointId();
+        final File integrityDir = new File(getIntegrityDataPath(endpointId));
+        if (!integrityDir.exists()) {
+            integrityDir.mkdir();
+        }
+
+        final File statusFile = new File(getIntegrityDataFile(endpointId, INTEGRITY_DATA_STATUS_FILENAME));
+        if (statusFile.exists()) {
+            statusFile.delete();
+        }
+
+        final Properties statusData = new Properties();
+        final BiConsumer<String, String> addData = (data, name) -> {
+            if (StringUtils.isNotBlank(data)) {
+                statusData.setProperty(name, data);
+            }
+        };
+
+        addData.accept(endpointId, REQUESTER_ENDPOINT);
+        addData.accept(integrityDataControl.getRequestId(), INTEGRITY_DATA_REQUEST_ID);
+        statusData.setProperty(INTEGRITY_DATA_STATUS, integrityDataControl.getProcessStatus().toString().toUpperCase());
+        addData.accept(integrityDataControl.getErrorMessage(), INTEGRITY_DATA_ERROR_MESSAGE);
+
+        try (FileOutputStream output = new FileOutputStream(statusFile)) {
+            statusData.store(output, null);
+        } catch (IOException e) {
+            throw new DotExecutionException(
+                    String.format("Could not save status to %s", statusFile.getAbsolutePath()),
+                    e);
+        }
+    }
+
+    public static void saveIntegrityDataStatus(final String endpointId,
+                                               final String requestId,
+                                               final IntegrityResource.ProcessStatus processStatus,
+                                               final String errorMessage) {
+        saveIntegrityDataStatus(new IntegrityDataControl(endpointId, requestId, processStatus, errorMessage));
+    }
+
+    public static void saveIntegrityDataStatus(final String endpointId,
+                                               final String requestId,
+                                               final IntegrityResource.ProcessStatus processStatus) {
+        saveIntegrityDataStatus(endpointId, requestId, processStatus, null);
+    }
+
+    public static void resetIntegrityDataControl(final String endpointId, final String requestId) {
+
+    }
+
     /**
      * Creates all the CSV from End Point database table and store them inside
      * zip file.
      *
      * @param endpointId
      * @throws Exception
      */
-    public void generateDataToCheckZip(String endpointId) throws Exception {
-        File zipFile = null;
-
-        try {
-            if (!UtilMethods.isSet(endpointId))
-                return;
-
-            final String outputPath = ConfigUtils.getIntegrityPath() + File.separator + endpointId;
+    public static void generateDataToCheckZip(final String endpointId) throws Exception {
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6151243)